.github/workflows/pre-post-release.yml

```diff
@@ -100,3 +92,5 @@ jobs:
 
       - name: Create the ${{ inputs.phase }} release PR
         run: gh pr create --title="${{ inputs.pr_title }}" --base main --head ${{ env.BRANCH_NAME }} -b "${{ inputs.pr_body }}"
+        env:
```

Comment:
IIUC, this is needed, otherwise gh will fail if the env variable is not set

Reply:
according to https://cli.github.com/manual/gh_help_environment it's not needed.

gh cli will first look for `GH_TOKEN` and then for `GITHUB_TOKEN`

Reply:
but for this special usecase, would the `${{ github.token }}` variable suffice?

Reply:
Ah, nevermind. Using the actions token might cause the tests to not run on the PR.

Reply:
Aha, that's the reason it worked because the `env.GITHUB_TOKEN` was set by `elastic/apm-pipeline-library/.github/actions/github-token@current`.

I somehow prefer to be explicit with the GH_TOKEN env variable in the step itself, I've seen a few broken builds  because it was not set properly.

That's correct, use a PAT/Finer-grained token should help with the events and builds